src/types/iterator.rs

```diff
@@ -52,6 +52,31 @@ impl PyIterator {
     }
 }
 
+#[derive(Debug)]
+pub enum PySendResult<'py> {
+    Next(Bound<'py, PyAny>),
+    Return(Bound<'py, PyAny>),
+}
+
+impl<'py> Bound<'py, PyIterator> {
+    /// Sends a value into the iterator.
```

Comment:
> I say generator above... I guess this function only makes sense for generators? (As far as I know, `send` is always paired with the `yield` keyword in Python? I wonder, is there a way we can replicate the `yield` in Rust?)

We cannot distinguish between normal python iterators and generators (because they are the same). A normal iterator just ignores the value being send.